docs/Custom_Device_Tips_and_Tricks.md

```diff
@@ -0,0 +1,271 @@
+•	All Ptrs—Array of references to the items the user pasted. You can only select one item to copy. This array only contains one reference that matches the Ptr in reference.<br />
+
+**ActionVIOnCompile**
+Executes when VeriStand compiles the system definition file.
+If you deploy the system definition, then undeploy it, and then redeploy it without making changes, this template does not execute because the system definition does not recompile.
+
+You can customize this template to finish configuring your hardware. The system definition file compiles when a user deploys the system definition, so you can configure your hardware based on the final settings from the system definition.
+
+You can also customize the template to quickly gather host-side settings. For example, often the custom device RT Engine VI uses properties set in the system definition. You can customize this template to read the values on the host side, which is much faster than reading them from the real-time target. You can then gather the properties into a single cluster, convert that cluster to a data variant, and write the variant as a single item property.
+
+### Adding Toolbar buttons
+
+A *Toolbar button* appears in the toolbar of **System Explorer**. These buttons only appear when displaying the configuration page associated with the button.
+
+Within the **&lt;Page&gt;** tags for an item, you can use the **&lt;ButtonList&gt;** tag to configure the toolbar buttons that appear with the item's configuration page. Each **&lt;Button&gt;** must include a unique **&lt;ID&gt;** string that identifies the button. The toolbar button displays by default.
+
+However, in each page VI, you can use the Disable Dynamic Button VI and the Enable Dynamic Button VI to dynamically disable and enable a button for that page based on its unique ID. These VIs are useful when you want the toolbar button to appear only when certain conditions are true.
+
+These VIs are located in the labview\vi.lib\NI VeriStand\Custom Device API directory.
+
+The following is an example framework you can use to implement a toolbar button.
+
+![](images/Custom_Device_Dynamic_Button_Framework.JPG)<br />
+**Custom Device Toolbar button framework**
+
+### Adding shortcut menus
+
+A *shortcut menu* for an item is the menu that appears when you right-click the item in System Explorer.
+
+Within the **&lt;Page&gt;** tags for an item, you can use the **&lt;RunTimeMenu&gt;** tag to configure the shortcut menu for the item. Each **&lt;MenuItem&gt;** you add under **&lt;RunTimeMenu&gt;** includes an **&lt;Item2Launch&gt;** section that specifies a VI to run when an operator selects the menu item.
+
+The Custom Device API library includes a template for this VI, RunTimeMenu Custom Item 2 Launch.vit, in the labview\vi.lib\NI VeriStand\Custom Device API directory.
+
+The following is an example framework you can use to implement a shortcut menu.
+
+![](images/Custom_Device_XML_Right-Click_Framework.JPG)<br />
```

Comment:
This picture needs to be updated also.
How it looks:
![image](https://user-images.githubusercontent.com/62251693/135118154-7a2fadf9-8ae8-48d6-8df0-2d3ef4197298.png)


How it should look like:
![image](https://user-images.githubusercontent.com/62251693/135118033-9d787526-7c06-46e7-93ee-9d3bacc18aa7.png)